skribenten-web/frontend/src/Brevredigering/LetterEditor/actions/toggleBulletList.ts

```diff
@@ -0,0 +1,566 @@
+import type { Draft } from "immer";
+import { produce } from "immer";
+import { isEqual } from "lodash";
+
+import type { Content, ItemList, TextContent } from "~/types/brevbakerTypes";
+
+import type { Action } from "../lib/actions";
+import type { LetterEditorState } from "../model/state";
+import { newItemList, newParagraph } from "./common";
+import type { ItemContentIndex, LiteralIndex } from "./model";
+
+export const toggleBulletList: Action<LetterEditorState, [literalIndex: LiteralIndex]> = produce(
+  (draft, literalIndex) => {
+    const block = draft.redigertBrev.blocks[literalIndex.blockIndex];
+    if (block.type !== "PARAGRAPH") {
+      return;
+    }
+
+    draft.isDirty = true;
+    const theContentTheUserIsOn = block.content[literalIndex.contentIndex];
+    if (theContentTheUserIsOn.type === "LITERAL" || theContentTheUserIsOn.type === "VARIABLE") {
+      toggleBulletListOn(draft, literalIndex);
+    } else if (theContentTheUserIsOn.type === "ITEM_LIST") {
+      toggleBulletListOff(draft, literalIndex as ItemContentIndex);
+    }
+  },
+);
+
+/**
+ * Når vi lager et punkt, så må vi ta høyde for at det kan være en punktliste før, etter, eller ingen punktliste.
+ *  Hvis det finnes en punktlisten før/etter/begge, må vi merge blockene til kun 1 block.
+ *
+ * Fordi vi gjør en såpass stor endring i dokument strukturen, Så må vi oppdatere fokuset til editorstaten til å være på rett plass
+ */
+const toggleBulletListOn = (draft: Draft<LetterEditorState>, literalIndex: LiteralIndex) => {
+  const blockBeforeThisblock = draft.redigertBrev.blocks[literalIndex.blockIndex - 1];
```

Comment:
Jeg tror vi må tenke på content som potensielt er før gjeldende content i samme blokk også. Så det som må sjekkes for er om det er en itemlist som umiddelbar nabo enten i samme blokk eller som siste/første i nabo-blokk. Gjelder både for før og etter.

Reply:
Yep - det har visst vært buggy med content før/etter. Disse skal nå være håndtert (ihvertfall når man toggler av. Må ta en en sjekk på når man toggler på)

Reply:
Slik jeg leser koden her, så vil det for toggling på skje som jeg beskrev (igorere content før gjeldende i samme blokk når lister slåes sammen).

Reply:
Da skal test-casene gå grønnt. Derimot så ser det ut som at jeg kanskje har introdusert en ny bug. Men jeg mener at jeg håndterer deleted content/blocks på riktig måte.